lib/Support/JitRunner.cpp

```diff
@@ -308,8 +308,8 @@ int mlir::JitRunnerMain(
     if (failed(mlirTransformer(m.get())))
       return EXIT_FAILURE;
 
-  auto transformer =
-      mlir::makeLLVMPassesTransformer(passes, optLevel, optPosition);
+  auto transformer = mlir::makeLLVMPassesTransformer(
+      passes, optLevel, /*targetMachine=*/nullptr, optPosition);
```

Comment:
Here can (and should?) we do something like:

```
auto TM_or_error = JITTargetMachineBuilder::detectHost().createTargetMachine();
if (!TM) return EXIT_FAILURE;
auto transformer = mlir::makeLLVMPassesTransformer(
		       passes, optLevel, /*targetMachine=*/TM_or_error->get(), optPosition);
```

Reply:
No strong opinion here but maybe we should add this in a separate commit since this is changing runner's behavior. Also, I'm not sure how much we are getting from `detectHost().createTargetMachine()` since this is not adding sub-target features. We would need something more in the direction of what I pointed out (https://github.com/dcaballe/mlir/pull/1) but I understood you preferred that code in LLVM. Let me know if we can reuse part of that code. I could prepare another commit with that. WDYT?

Reply:
Isn't most of the TLI independent of sub-target features?
In any case adding sub-target features seems like the kind of thing that `detectHost().createTargetMachine()` should take care of, don't you think?

Reply:
Vectorization is sub-target dependent, for example.

I agree with you but I feel there must be a reason why it's not there already :). Let me try to add it to LLVM and I'll create another PR to update jit runner if you don't beat me to it.